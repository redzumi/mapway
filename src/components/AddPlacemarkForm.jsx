import PropTypes from 'prop-types'
import React, { Component } from 'react'
import styled from 'styled-components'

const DEFAULT_PLACEMARK = { name: '' }

class AddPlacemarkForm extends Component {
  constructor(props) {
    super(props)
    this.state = { formData: DEFAULT_PLACEMARK }
  }

  handleChange = e => {
    this.setState({ formData: { ...this.state.formData, [e.target.name]: e.target.value } })
  }

  handleSubmit = e => {
    e.preventDefault()
    this.props.onPlacemarkAdd(this.state.formData)
    this.setState({ formData: DEFAULT_PLACEMARK })
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <StyledInput
          name="name"
          required
          placeholder="Новая точка маршрута"
          value={this.state.formData.name}
          onChange={this.handleChange}
        />
      </form>
    )
  }
}

AddPlacemarkForm.propTypes = {
  onPlacemarkAdd: PropTypes.func.isRequired,
}

const StyledInput = styled.input`
  background-color: #fff;
  width: 300px;
  font-family: 'Montserrat', sans-serif;
  padding-left: 10px;
  height: 60px;
  margin: 0px;
  outline: none;
  box-sizing: border-box;
  border: 1px solid #efefef;
  transition: all 0.25s;
  margin-bottom: 20px;
  &:focus {
    border: 1px solid #a7a7a7;
  }
`

export { AddPlacemarkForm as default, StyledInput }
