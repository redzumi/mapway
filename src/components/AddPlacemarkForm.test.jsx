import React from 'react'
import { shallow } from 'enzyme'
import AddPlacemarkForm, { StyledInput } from './AddPlacemarkForm'

describe('AddPlacemarkForm styled components', () => {
  it('renders input correctly', () => {
    expect(shallow(<StyledInput />)).toMatchSnapshot()
  })
})

describe('AddPlacemarkForm', () => {
  it('renders correctly', () => {
    expect(shallow(<AddPlacemarkForm onPlacemarkAdd={() => {}} />)).toMatchSnapshot()
  })

  it('simulates add placemark event', () => {
    const onPlacemarkAdd = jest.fn()
    const renderedComponent = shallow(<AddPlacemarkForm onPlacemarkAdd={onPlacemarkAdd} />)

    renderedComponent
      .find('[name="name"]')
      .simulate('change', { target: { name: 'name', value: 'Test placemark name' } })
    renderedComponent.find('form').simulate('submit', { preventDefault() {} })

    expect(onPlacemarkAdd).toBeCalledWith({ name: 'Test placemark name' })
  })
})
