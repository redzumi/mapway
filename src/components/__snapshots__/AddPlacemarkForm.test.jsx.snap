// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlacemarkForm renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.input
    name="name"
    onChange={[Function]}
    placeholder="Новая точка маршрута"
    required={true}
    value=""
  />
</form>
`;

exports[`AddPlacemarkForm styled components renders input correctly 1`] = `
.c0 {
  background-color: #fff;
  width: 300px;
  font-family: 'Montserrat',sans-serif;
  padding-left: 10px;
  height: 60px;
  margin: 0px;
  outline: none;
  box-sizing: border-box;
  border: 1px solid #efefef;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  margin-bottom: 20px;
}

.c0:focus {
  border: 1px solid #a7a7a7;
}

<input
  className="c0"
/>
`;
